config/jobs/kubernetes-csi/gen-jobs.sh

```diff
@@ -558,7 +562,7 @@ EOF
   - name: $(job_name "pull" "$repo" "$tests")
     cluster: $(job_cluster "$repo")
     always_run: true
-    optional: $(pull_optional "$tests")
+    optional: $(pull_optional "$tests" "$kubernetes" "$deployment_suffix" "$repo")
```

Comment:
Add a TODO to remove those added parameters when no longer needed.

Reply:
You also added "$kubernetes" "$deployment_suffix" here.  Are they needed for this change?